### PR TITLE
Release internal 1.4.0

### PR DESCRIPTION
**Pull request by benbp**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@benbp now that you're back, do we need to resurrect this?

**Comment by benbp** [comment]
@jhendrixMSFT yep! I was going to reach out on Monday.

**Comment by RickWinter** [comment]
ping: @benbp , @jhendrixMSFT

File: sdk/internal/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.3.1 (Unreleased)
+## 1.4.0 (2023-07-27)
```
Comment (RickWinter) [inline sdk/internal/CHANGELOG.md:3]: The date needs updating

**Comment by jhendrixMSFT** [comment]
I approved the APIView https://apiview.dev/Assemblies/Review/a6e1eabcfada4b75b6a8b465be2fe24b

File: sdk/internal/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 
 ### Features Added
```
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/internal/CHANGELOG.md:5]: fixed

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**